warn-stranded.lua

```diff
@@ -0,0 +1,204 @@
+warning.ATTRS = {
+    focus_path='warn-stranded',
+    force_pause=true,
+    pass_mouse_clicks=false,
+}
+
+function warning:init(info)
+    self:addviews{
+       widgets.Window{
+          view_id = 'main',
+          frame={w=80, h=18},
+          frame_title='Stranded Citizen Warning',
+          resizable=true,
+          subviews = {
+             widgets.Label{
+                frame = { l = 0, t = 0},
+                text_pen = COLOR_CYAN,
+                text = 'Number Stranded: '..#info.units,
+             },
+             widgets.List{
+                view_id = 'list',
+                frame = { t = 3, b = 5 },
+                text_pen = { fg = COLOR_GREY, bg = COLOR_BLACK },
+                cursor_pen = { fg = COLOR_BLACK, bg = COLOR_GREEN },
+             },
+             widgets.Label{
+                view_id = 'bottom_ui',
+                frame = { b = 0, h = 1 },
+                text = 'filled by updateBottom()'
+             }
+          }
+       }
+    }
+
+    self:initListChoices(info.units)
+    self:updateBottom()
+end
+
+local function getSexString(sex)
+  local sym = df.pronoun_type.attrs[sex].symbol
+  if not sym then
+    return ""
+  end
+  return "("..sym..")"
+end
+
+local function getUnitDescription(unit)
+   return '['..dfhack.units.getProfessionName(unit)..'] '..dfhack.TranslateName(dfhack.units.getVisibleName(unit))..
+      ' '..getSexString(unit.sex)..' Stress category: '..dfhack.units.getStressCategory(unit)
+end
+
+
+local function unitIgnored(unit)
+   local currentIgnore = dfhack.persistent.get('warnStrandedIgnore')
+   if currentIgnore == nil  then return false end
+
+   for index, id in pairs(currentIgnore['ints']) do
+      if id == unit.id then
+         print(id)
+         print(unit.id)
+         print('Unit Ignored')
+         return true, index
+      end
+   end
+
+   return false
+end
+
+local function toggleUnitIgnore(unit)
+   local currentIgnore = dfhack.persistent.get('warnStrandedIgnore')
+
+   if currentIgnore == nil then
+      currentIgnore = { key = 'warnStrandedIgnore', value = 'warnStrandedIgnore', ints = {} }
+      dfhack.printerr('nil, creating')
+      table.insert(currentIgnore['ints'], unit.id)
+   end
+
+   local ignored, index = unitIgnored(unit)
+
+   if ignored then
+      dfhack.printerr('toggle remove')
+      table.remove(currentIgnore['ints'], unit.id)
+   else
+      dfhack.printerr('toggle insert')
+      table.insert(currentIgnore['ints'], unit.id)
+   end
+
+   dfhack.persistent.save(currentIgnore)
+end
+
+function warning:initListChoices(units)
+   local choices = {}
+   for _, unit in pairs(units) do
+      table.insert(choices, { text = getUnitDescription(unit), unit = unit })
+   end
+   local list = self.subviews.list
+   list:setChoices(choices, 1)
+end
+
+function warning:updateBottom()
+   self.subviews.bottom_ui:setText(
+      {
+         { key = 'SELECT', text = ': Toggle ignore unit', on_activate = self:callback('onIgnore') }, ' ',
```

Comment:
please use `HotkeyLabel` widgets instead of hand-coding these "buttons". I realize you probably got this from `gui/autobutcher`, but that code is ancient and needs to be updated to modern standards. `gui/control-panel` would probably be a better model.